psl/parser-database/src/walkers/relation.rs

```diff
@@ -38,16 +38,28 @@ impl<'db> RelationWalker<'db> {
 
     /// Is this a relation where both ends are the same model?
     pub fn is_self_relation(self) -> bool {
-        let r = self.get();
+        let r = self.ast_relation();
         r.model_a == r.model_b
     }
 
+    /// Gets relation kind
+    pub fn relation_kind(self) -> &'db str {
+        let r = self.ast_relation();
+
+        match r.attributes {
+            RelationAttributes::ImplicitManyToMany { .. } => "implicit many-to-many",
+            RelationAttributes::TwoWayEmbeddedManyToMany { .. } => "explicit many-to-many",
```

Comment:
I'm a bit torn on this one. I don't think `TwoWayEmbeddedManyToMany` qualifies as explicit. They refer to a specific kind of implicit m2m for MongoDB. `"embedded many-to-many"` won't make much sense for users though. So I'm not too opinionated on this but I would consider referring to both `ImplicitM2M` and `TwoWayEmbeddedM2M` as `"implicit many-to-many"`.

Reply:
Ah, hm 🤔 thanks for clarifying!